File: CS/WindowsApplication3/Main.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.Skins.XtraForm;
using DevExpress.Skins;
using DevExpress.Utils;
using System.Drawing.Drawing2D;
using DevExpress.Utils.Drawing;
using DevExpress.Utils.Drawing.Helpers;


namespace DXSample {
    public partial class Main: XtraForm {
        public Main() {
            InitializeComponent();
        }

        protected override DevExpress.Skins.XtraForm.FormPainter CreateFormBorderPainter()
        {
            return new MyFormPainter(this, DevExpress.LookAndFeel.UserLookAndFeel.Default.ActiveLookAndFeel);
        }
    }

    public class MyFormPainter : FormPainter
    {
        int x = 0, delta = 8;
        Image image;
        Timer timer;

        public MyFormPainter(Control owner, ISkinProvider provider) : base(owner, provider) {
            timer = new Timer();
            timer.Interval = 300;
            timer.Tick += OnTick;
            timer.Start();
        }
       
        Image Image {
            get
            {
                if (image == null)
                    image = CreateImage();
                return image;
            }
        }
        private Bitmap CreateImage()
        {
            Bitmap temp = new Bitmap(TextBounds.Width, TextBounds.Height);
            using (Graphics gr = Graphics.FromImage(temp))
            {
                for (int i = 0; i < temp.Width; i += delta)
                    for (int j = 0; j < temp.Height; j += delta)
                    {
                        Rectangle rect = new Rectangle(i, j, delta, delta);
                        LinearGradientBrush brush = new LinearGradientBrush(rect, Color.White, Color.Black, LinearGradientMode.BackwardDiagonal);
                        LinearGradientBrush borderBrush = new LinearGradientBrush(rect, Color.DarkGray, Color.White, LinearGradientMode.BackwardDiagonal);
                        Pen pen = new Pen(borderBrush);
                        gr.FillRectangle(brush, rect);
                        gr.DrawRectangle(pen, rect);
                    }
            }
            return temp;
        }

        void OnTick(object sender, EventArgs e)
        {
            if (x <= TextBounds.Right - delta)
                x += delta;
            else
                x = 0;
            UpdateText();
        }

        private void UpdateText()
        {
           DrawFrameNC(new Message());
        }

        protected override void DrawText(DevExpress.Utils.Drawing.GraphicsCache cache)
        {
            if (Text == null || Text.Length == 0 || TextBounds.IsEmpty) return;
            AppearanceObject appearance = new AppearanceObject(GetDefaultAppearance());
            cache.Graphics.DrawImage(Image, TextBounds);
            Rectangle textRect = new Rectangle(TextBounds.X + x, TextBounds.Y, TextBounds.Width - x, TextBounds.Height);
            DrawTextShadow(cache, appearance, textRect);
            cache.Graphics.DrawString(Text, appearance.Font, Brushes.WhiteSmoke, textRect, new StringFormat(StringFormatFlags.NoWrap));
        }

        public override void Dispose()
        {
            timer.Stop();
            timer.Tick -= OnTick;
            timer.Dispose();
            base.Dispose();
        }
    }
}

File: VB/WindowsApplication3/Main.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.Skins.XtraForm
Imports DevExpress.Skins
Imports DevExpress.Utils
Imports System.Drawing.Drawing2D
Imports DevExpress.Utils.Drawing
Imports DevExpress.Utils.Drawing.Helpers

Namespace DXSample

    Public Partial Class Main
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub

        Protected Overrides Function CreateFormBorderPainter() As FormPainter
            Return New MyFormPainter(Me, DevExpress.LookAndFeel.UserLookAndFeel.Default.ActiveLookAndFeel)
        End Function
    End Class

    Public Class MyFormPainter
        Inherits FormPainter

        Private x As Integer = 0, delta As Integer = 8

        Private imageField As Image

        Private timer As Timer

        Public Sub New(ByVal owner As Control, ByVal provider As ISkinProvider)
            MyBase.New(owner, provider)
            timer = New Timer()
            timer.Interval = 300
            AddHandler timer.Tick, AddressOf OnTick
            timer.Start()
        End Sub

        Private ReadOnly Property Image As Image
            Get
                If imageField Is Nothing Then imageField = CreateImage()
                Return imageField
            End Get
        End Property

        Private Function CreateImage() As Bitmap
            Dim temp As Bitmap = New Bitmap(TextBounds.Width, TextBounds.Height)
            Using gr As Graphics = Graphics.FromImage(temp)
                Dim i As Integer = 0
                While i < temp.Width
                    Dim j As Integer = 0
                    While j < temp.Height
                        Dim rect As Rectangle = New Rectangle(i, j, delta, delta)
                        Dim brush As LinearGradientBrush = New LinearGradientBrush(rect, Color.White, Color.Black, LinearGradientMode.BackwardDiagonal)
                        Dim borderBrush As LinearGradientBrush = New LinearGradientBrush(rect, Color.DarkGray, Color.White, LinearGradientMode.BackwardDiagonal)
                        Dim pen As Pen = New Pen(borderBrush)
                        gr.FillRectangle(brush, rect)
                        gr.DrawRectangle(pen, rect)
                        j += delta
                    End While

                    i += delta
                End While
            End Using

            Return temp
        End Function

        Private Sub OnTick(ByVal sender As Object, ByVal e As EventArgs)
            If x <= TextBounds.Right - delta Then
                x += delta
            Else
                x = 0
            End If

            UpdateText()
        End Sub

        Private Sub UpdateText()
            DrawFrameNC(New Message())
        End Sub

        Protected Overrides Sub DrawText(ByVal cache As DevExpress.Utils.Drawing.GraphicsCache)
            If Equals(Text, Nothing) OrElse Text.Length = 0 OrElse TextBounds.IsEmpty Then Return
            Dim appearance As AppearanceObject = New AppearanceObject(GetDefaultAppearance())
            cache.Graphics.DrawImage(Image, TextBounds)
            Dim textRect As Rectangle = New Rectangle(TextBounds.X + x, TextBounds.Y, TextBounds.Width - x, TextBounds.Height)
            DrawTextShadow(cache, appearance, textRect)
            cache.Graphics.DrawString(Text, appearance.Font, Brushes.WhiteSmoke, textRect, New StringFormat(StringFormatFlags.NoWrap))
        End Sub

        Public Overrides Sub Dispose()
            timer.Stop()
            RemoveHandler timer.Tick, AddressOf OnTick
            timer.Dispose()
            MyBase.Dispose()
        End Sub
    End Class
End Namespace
